synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1009,3 +1043,75 @@ def purged_chain_cover_txn(txn) -> int:
             await self.db_pool.updates._end_background_update("purged_chain_cover")
 
         return result
+
+    async def _background_populate_stream_ordering2(
+        self, progress: JsonDict, batch_size: int
+    ) -> int:
+        """Populate events.stream_ordering2, then replace stream_ordering
+
+        This is to deal with the fact that stream_ordering was initially created as a
+        32-bit integer field.
+        """
+        batch_size = max(batch_size, 1)
```

Comment:
Why is this an issue here? I'd expect setting batch_size to zero/negative anywhere is going to cause issues?

Reply:
true. It's just defensive here.